src/Uno.UI.Runtime.Skia.Android/Microsoft/UI/Xaml/UnoSKCanvasView.cs

```diff
@@ -272,6 +305,16 @@ void IRenderer.OnDrawFrame(IGL10? gl)
 			// else : we already drew directly on the OpenGL-backed canvas
 
 			_context!.Flush();
+
+			if (!CompositionTarget.IsRenderingActive && ReferenceEquals(currentPicture, Volatile.Read(ref surfaceView._picture)))
```

Comment:
Same question. Also I'm not sure why the picture comparison is necessary. If the picture changed, this means that we should've invalidated the surface view and OnDraw will get called again soon enough.

Reply:
The picture comparison is necessary for race conditions, where the image may have been changed in between the beginning of the frame and the check we make, as the frame may have been requested manually for pointers or others.